pkg/sanity/volume.go

```diff
@@ -0,0 +1,194 @@
+/*
+Copyright 2017 Portworx
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sanity
+
+import (
+	"strconv"
+
+	"github.com/libopenstorage/openstorage/api"
+	"github.com/libopenstorage/openstorage/api/client"
+	volumeclient "github.com/libopenstorage/openstorage/api/client/volume"
+
+	"github.com/libopenstorage/openstorage/volume"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Volume [Volume Tests]", func() {
+	var (
+		restClient   *client.Client
+		volumedriver volume.VolumeDriver
+	)
+
+	BeforeEach(func() {
+		var err error
+		restClient, err = volumeclient.NewDriverClient(osdAddress, "nfs", volume.APIVersion, "")
```

Comment:
You cannot specify here the driver. It must be specified by the caller. If a driver needs to be passed in, adjust sanity.go to take in a driver name.

The reason is that this would make this program only test NFS, while instead we want it to test the specific implementation running.